bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase1_x/TestMetrics.java

```diff
@@ -0,0 +1,231 @@
+    configuration.set(BigtableOptionsFactory.BIGTABLE_USE_GCJ_CLIENT, "true");
+    connection = new BigtableConnection(configuration);
+  }
+
+  @BeforeClass
+  public static void setUpServer() throws IOException {
+    try (ServerSocket s = new ServerSocket(0)) {
+      dataPort = s.getLocalPort();
+    }
+    server = ServerBuilder.forPort(dataPort).addService(fakeDataService).build();
+    server.start();
+  }
+
+  @AfterClass
+  public static void tearDownServer() throws InterruptedException {
+    if (server != null) {
+      server.shutdownNow();
+      server.awaitTermination();
+    }
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    connection.close();
+  }
+
+  @Test
+  public void readRows() throws IOException, InterruptedException {
+    FakeMetricRegistry fakeMetricRegistry = new FakeMetricRegistry();
+
+    BigtableClientMetrics.setMetricRegistry(fakeMetricRegistry);
+    Table table = connection.getTable(TABLE_NAME);
+
+    long readRowsStart = System.currentTimeMillis();
+    Result result = table.get(new Get(new byte[2]));
+    long readRowsTime = System.currentTimeMillis() - readRowsStart;
+
+    ReadRowsRequest request = fakeDataService.popLastRequest();
+    assertEquals(FULL_TABLE_NAME, request.getTableName());
+
+    AtomicLong readRowFailure =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.grpc.method.ReadRow.failure");
+    AtomicLong operationLatency =
+        fakeMetricRegistry.results.get(
+            "google-cloud-bigtable.grpc.method.ReadRow.operation.latency");
+    AtomicLong tableGetLatency =
+        fakeMetricRegistry.results.get("google-cloud-bigtable.table.get.latency");
+
+    Assert.assertEquals(3, readRowFailure.get());
+    Assert.assertTrue(
+        "operation latency for ReadRow took longer than expected",
+        operationLatency.get() <= readRowsTime + 50);
+    Assert.assertTrue(
+        "operation latency for table.get took longer than expected",
+        tableGetLatency.get() <= readRowsTime + 50);
+  }
+
+  private static class FakeDataService extends BigtableGrpc.BigtableImplBase {
+
+    final BlockingQueue<Object> requests = Queues.newLinkedBlockingDeque();
+
+    @SuppressWarnings("unchecked")
+    <T> T popLastRequest() throws InterruptedException {
+      return (T) requests.poll(1, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void readRows(
+        ReadRowsRequest request, StreamObserver<ReadRowsResponse> responseObserver) {
+      requests.add(request);
+      if (callCount.getAndIncrement() < 4) {
+        responseObserver.onError(new StatusRuntimeException(Status.UNAVAILABLE));
+      } else {
+        responseObserver.onNext(ReadRowsResponse.newBuilder().build());
+        responseObserver.onCompleted();
+      }
+    }
+  }
+
+  private static class FakeMetricRegistry implements MetricRegistry {
+
+    private final Map<String, AtomicLong> results = new HashMap<>();
```

Comment:
I think this will cause flakes because the registry will be accessed from multiple threads.  I think you might be better off with using synchronization here:

```java
private final Object lock = new Object();
Map<String, Long> stats = new HashMap();

inc() {
  synchronized(lock) {
    long count = Objects.firstNonNull(stats.get(name), 0);
    count++;
    stats.put(name, count);
  }
}
```